src/CodeReport/SourceCodeReportTest.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock-actions.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "ClientData/PostProcessedSamplingData.h"
+#include "CodeReport/SourceCodeReport.h"
+#include "ObjectUtils/ObjectFile.h"
+#include "capture_data.pb.h"
+#include "symbol.pb.h"
+
+namespace {
+class MockElfFile : public orbit_object_utils::ElfFile {
+ public:
+  MOCK_METHOD(ErrorMessageOr<orbit_grpc_protos::ModuleSymbols>, LoadSymbolsFromDynsym, (),
+              (override));
+  MOCK_METHOD(ErrorMessageOr<uint64_t>, GetLoadBias, (), (const, override));
+
+  MOCK_METHOD(bool, HasDynsym, (), (const, override));
+  MOCK_METHOD(bool, HasDebugInfo, (), (const, override));
+  MOCK_METHOD(bool, HasGnuDebuglink, (), (const, override));
+  MOCK_METHOD(bool, Is64Bit, (), (const, override));
+  MOCK_METHOD(std::string, GetSoname, (), (const, override));
+  MOCK_METHOD(std::string, GetBuildId, (), (const, override));
+  MOCK_METHOD(ErrorMessageOr<orbit_grpc_protos::LineInfo>, GetLineInfo, (uint64_t), (override));
+  MOCK_METHOD(ErrorMessageOr<orbit_grpc_protos::LineInfo>, GetDeclarationLocationOfFunction,
+              (uint64_t), (override));
+  MOCK_METHOD(std::optional<orbit_object_utils::GnuDebugLinkInfo>, GetGnuDebugLinkInfo, (),
+              (const, override));
+
+  MOCK_METHOD(ErrorMessageOr<orbit_grpc_protos::ModuleSymbols>, LoadDebugSymbols, (), (override));
+  MOCK_METHOD(bool, HasDebugSymbols, (), (const, override));
+  MOCK_METHOD(std::string, GetName, (), (const, override));
+  MOCK_METHOD(const std::filesystem::path&, GetFilePath, (), (const, override));
+  MOCK_METHOD(bool, IsElf, (), (const, override));
+  MOCK_METHOD(bool, IsCoff, (), (const, override));
+};
+}  // namespace
+
+namespace orbit_code_report {
+TEST(SourceCodeReport, Empty) {
+  orbit_client_protos::FunctionInfo function_info{};
+  function_info.set_pretty_name("main()");
+  function_info.set_address(0x42);
+  function_info.set_size(0x100);
+
+  MockElfFile elf_file{};
+  EXPECT_CALL(elf_file, GetLineInfo).Times(0);
+
+  orbit_client_data::ThreadSampleData sample_data{};
+
+  SourceCodeReport report{"", function_info, 0x8000, &elf_file, sample_data, 0ul};
+
+  EXPECT_EQ(report.GetNumSamples(), 0);
+  EXPECT_EQ(report.GetNumSamplesInFunction(), 0);
+  EXPECT_FALSE(report.GetNumSamplesAtLine(0).has_value());
+}
+
+TEST(SourceCodeReport, StaticLineInfo) {
```

Comment:
Likely a stupid question, but what does "static" mean in this context?

Reply:
Not a stupid question. Makes not really sense since everything is static. I will change it.

Reply:
Done.